pymeasure/instruments/hp/hphelper.py

```diff
@@ -0,0 +1,260 @@
+#
```

Comment:
My proposal is to add in this common file, the support for managing bitfield and byte and remove instrument specific information.
In order word, you could create a class (let's call for the sake of discussion `HPLegacyInstrument`) inheriting from `Instrument` with static methods implementing the factories.
Moving the specific instrument's definition and initialization to specific instrument file.
Specific instrument classes are now inheriting from `HPLegacyInstrument`.
I noticed also that `__float__` method seems specific to HP3437A. Is this the case ?
I have to say that common part is not so big, so, people may argue that at the end it may not be worth having a separate file. Also some people don't like long chain of inheritance

Reply:
Thank you for the further explanation/proposal, that makes much more sense, then the way I tried with adding elements from an external class.

The `__float__`method is specific for the HP3437A at the mooment, as the HP3478A does not use fractional numbers encoded in BCD.